### DISCUSSION TITLE
/back throws an exception and returns "No valid location found."

### DISCUSSION THREAD

**Discussion by ZionFox** [EssentialsX/Essentials#5515] [Q&A]
To preface: Server plugins was migrated from Essentials 2.17.1.9. `plugins/Essentials` and most flat files within it are from that version. I did follow migration guide as to 'delete' the config.yml and let EssentialsX recreate it. I did not see that EssX has changed any of the userdata files at all, as in there has been zero change to the _modified_ timestamps.

Server dump available here: https://essentialsx.net/dump.html?id=5c6bb3833a024e89890b39cc11c5beda

I have enabled the 'let essentials hook teleports' setting in config.yml to no avail.

In the older version of Essentials, there was a `lastlocation` object which stored the player's last position, be it from teleport, or death. I believe this data was used by Essentials for the `/back` command. This doesn't seem to be getting populated since updating to EssX, and for the one user who already has one (myself), the `lastlocation:` object isn't being updated at all.

Below is the UUID.yml for `plugins/Essentials/userdata/` for myself:
```timestamps:
  login: 1694658145199
  logout: 1694569091418
  lastteleport: 1694661059733
  lastheal: 0
  mute: 0
  jail: 0
  onlinejail: 0
afk: true
logoutlocation:
  world: 596a8f96-487e-4c18-94d2-4476e61bf954
  x: -2.3964779500040896E7
  y: 94.0
  z: 2.5283521028552007E7
  yaw: -1.91748046875
  pitch: 3.750380039215088
  world-name: hawknet
godmode: false
lastlocation:
  world: skyworld
  x: -2.947237505550377
  y: -71.99353744489679
  z: -2.157174449705336
  yaw: 0.0
  pitch: 0.0
  world-name: ''
homes:
  home:
    world: 596a8f96-487e-4c18-94d2-4476e61bf954
    x: -3527.8915567267995
    y: 64.0
    z: -951.8591559750996
    yaw: 82.142333984375
    pitch: 25.650800704956055
    world-name: hawknet
  skeletons:
    world: 596a8f96-487e-4c18-94d2-4476e61bf954
    x: -3582.4097526829228
    y: 47.0
    z: -944.507164237096
    yaw: 142.28060913085938
    pitch: 54.74964904785156
    world-name: hawknet
  mineshaft:
    world: 596a8f96-487e-4c18-94d2-4476e61bf954
    x: -3617.1660486963056
    y: 29.0
    z: -978.510202799219
    yaw: 90.36376953125
    pitch: 21.751314163208008
    world-name: hawknet
money: '263.74'
powertools:
  infested_stone:
  - /paste
  infested_cobblestone:
  - /replace grass sand
  red_tulip:
  - spawnentity skeleton
  orange_tulip:
  - lightning
ip-address: 172.22.0.1
last-account-name: ZionFox
teleportenabled: true
teleportauto: false
muted: false
jailed: false
socialspy: false
npc: false
powertoolsenabled: true
accepting-pay: true
baltop-exempt: false
```

As an asside, the only worlds available are titled `hawknet`, `hawknet_nether` and `hawknet_the_end`, the world `skyworld` in the `lastlocation:` object was from my MC 1.14.4 server, and controlled by Multiverse which is not present on this server.

**Comment by mdcfe** [comment]
2.17.1.9 is a very old EssentialsX version. Migration instructions are provided in each subsequent release changelog.

`/back` locations are stored in memory - they have never been persisted on disk. `lastlocation` was replaced by `logoutlocation`, and is used to display logout locations in `/whois` and allow server admins to teleport to offline players with `/tpoffline`. It has no relation to `/back`.

Could you verify whether players have the `essentials.back.ondeath` and `essentials.back.onteleport` permissions?

**Reply by ZionFox** [reply]
As noticed in the EssX dump, the permissions system I use is PermissionsEx, which is marked as being unsupported, but still seems to be working on the server without issues. The version is 1.23.4.

The setup in this means when a user first joins they're in the `Guest` group, and this specifically disables `essentials.back`. The next group after this, which is granted by an admin doing `/pex promote` specifically re-grants the following permissions (related to back)
```
essentials.back
essentials.back.ondeath
essentials.back.onteleport
essentials.back.into.*
```

All subsequent groups inherit these.

I know the permissions system is working, due to the fact that `/pex promote` works, and adding other permissions enables the commands to the user.
